### PR TITLE
Version 3.0

### PR DESCRIPTION
**Pull request by sepandhaghighi**
#### Reference Issues/PRs
#326 #330 
#### What does this implement/fix? Explain your changes.
Version 3.0

### CODE REVIEW

**Comment by sepandhaghighi** [comment]
@alirezazolanvari @sadrasabouri Hi folks
Please complete this review until **Monday - 2020-10-26 - 11:59 AM GMT+3:30**
**Notice:** Only review, don't need to merge

**Review by sadrasabouri — APPROVED** [review]

File: Document/Example2.ipynb [hunk 1]
```diff
@@ -14,6 +14,13 @@
     "# Example-2 (How to plot via matplotlib)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "⚠️ This example is deprecated. you can use `plot` method from `version 3.0`"
```
Comment (alirezazolanvari) [inline Document/Example2.ipynb:21]: Please change it to:
"⚠️ This example is deprecated. You can use `plot` method from `version 3.0`"

File: pycm/pycm_obj.py [hunk 2]
```diff
@@ -866,3 +866,87 @@ def to_array(self, normalized=False, one_vs_all=False, class_name=None):
             row = [table[key][i] for i in classes]
             array.append(row)
         return numpy.array(array)
+
+    def combine(self, other):
+        """
+        Return the combine of self Confusion matrix and other one.
```
Comment (alirezazolanvari) [inline pycm/pycm_obj.py:872]: Would you please reword this sentence? My suggestion is: "For combining the confusion matrix with another one."

File: pycm/pycm_obj.py [hunk 3]
```diff
@@ -866,3 +866,87 @@ def to_array(self, normalized=False, one_vs_all=False, class_name=None):
...
+
+        :param other: Other matrix that is going to be combined.
```
Comment (alirezazolanvari) [inline pycm/pycm_obj.py:874]: Please change it to:
:param other: The other matrix that is going to be combined.

File: pycm/pycm_obj.py [hunk 4]
```diff
@@ -866,3 +866,87 @@ def to_array(self, normalized=False, one_vs_all=False, class_name=None):
...
+        :type other: Confusion matrix
+        :return: combine of two matrices as a new Confusion matrix
```
Comment (alirezazolanvari) [inline pycm/pycm_obj.py:876]: Please change it to:
:return: The combination of two matrices as a new confusion matrix

File: pycm/pycm_util.py [hunk 5]
```diff
@@ -409,3 +409,111 @@ def statistic_recommend(classes, P):
     if binary_check(classes):
         return BINARY_RECOMMEND
     return MULTICLASS_RECOMMEND
+
+
+def matrix_combine(matrix_1, matrix_2):
+    """
+    Return the combine of two Confusion matrices.
```
Comment (alirezazolanvari) [inline pycm/pycm_util.py:416]: Return the combination of two confusion matrices.

File: pycm/pycm_param.py [hunk 6]
```diff
@@ -38,6 +40,8 @@
 
 COMPARE_WEIGHT_ERROR = "The weight type must be dictionary and also must be specified for all of the classes."
 
+COMBINE_TYPE_ERROR = "The input type is considered to be pycm.ConfusionMatrix object but it's not!"
```
Comment (alirezazolanvari) [inline pycm/pycm_param.py:43]: The input type is supposed to be pycm.ConfusionMatrix object but it's not!

File: pycm/pycm_util.py [hunk 7]
```diff
@@ -409,3 +409,111 @@ def statistic_recommend(classes, P):
...
+
+    :param matrix_1: First matrix that is going to be combined.
+    :type matrix_1: dict
+    :param matrix_2: Second matrix that is going to be combined.
+    :type matrix_2: dict
+    :return: combine of two matrices as a dict of dicts
```
Comment (alirezazolanvari) [inline pycm/pycm_util.py:422]: The combination of two matrices as a dict of dicts

**Review by alirezazolanvari — CHANGES_REQUESTED** [review]
Please apply the requested changes.

File: Document/Example2.ipynb (hunk 1 above)
Comment (sepandhaghighi) [reply Document/Example2.ipynb]: Fixed in https://github.com/sepandhaghighi/pycm/pull/344/commits/fe85fd4c1b8cde4f5e5f851b3a00fecbe8b7a0c6

File: pycm/pycm_util.py (hunk 7 above)
Comment (sepandhaghighi) [reply pycm/pycm_util.py]: Fixed in https://github.com/sepandhaghighi/pycm/pull/344/commits/b5eb3bc08c02b59bd9a452cfa786c0ca9dd412f0

File: pycm/pycm_util.py (hunk 5 above)
Comment (sepandhaghighi) [reply pycm/pycm_util.py]: Fixed in https://github.com/sepandhaghighi/pycm/pull/344/commits/b5eb3bc08c02b59bd9a452cfa786c0ca9dd412f0

File: pycm/pycm_obj.py (hunk 4 above)
Comment (sepandhaghighi) [reply pycm/pycm_obj.py]: Fixed in https://github.com/sepandhaghighi/pycm/pull/344/commits/b5eb3bc08c02b59bd9a452cfa786c0ca9dd412f0

File: pycm/pycm_obj.py (hunk 3 above)
Comment (sepandhaghighi) [reply pycm/pycm_obj.py]: Fixed in https://github.com/sepandhaghighi/pycm/pull/344/commits/b5eb3bc08c02b59bd9a452cfa786c0ca9dd412f0

File: pycm/pycm_obj.py (hunk 2 above)
Comment (sepandhaghighi) [reply pycm/pycm_obj.py]: Fixed in https://github.com/sepandhaghighi/pycm/pull/344/commits/b5eb3bc08c02b59bd9a452cfa786c0ca9dd412f0

File: pycm/pycm_param.py (hunk 6 above)
Comment (sepandhaghighi) [reply pycm/pycm_param.py:43]: This error message is exactly like `COMPARE_TYPE_ERROR`

Comment (alirezazolanvari) [reply pycm/pycm_param.py:43]: What's your opinion about changing both of them?

Comment (sepandhaghighi) [reply pycm/pycm_param.py]: Done!
Please take a look at https://github.com/sepandhaghighi/pycm/pull/344/commits/c9edcb62e5aae931d5b7231540857b64280083e2 and https://github.com/sepandhaghighi/pycm/pull/344/commits/2e4c543a6806dde36c3f709933f0f1fee2e0eefb

`COMPARE_FORMAT_ERROR` , `COMPARE_TYPE_ERROR`, `COMBINE_TYPE_ERROR`, `CI_FORMAT_ERROR` updated

Comment (sepandhaghighi) [reply pycm/pycm_param.py]: Also, take a look at #345

**Review by alirezazolanvari — APPROVED** [review]

**Pull request merged**